typescript.md

```diff
@@ -0,0 +1,73 @@
+TypeScript Code Style
+=====================
+
+## Table of Contents
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+- [General](#general)
+- [Naming](#naming)
+- [Typing](#typing)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## General
+
+This style guide is generally the same as the one for JavaScript.
+
+[&#8593; back to TOC](#table-of-contents)
+
+## Naming
+
+* `IInterfaceLikeThis` — notice `I` in the beginning.
+* In generics:
+  * `K` is reserved for key-like data types;
+  * `T` and `U` are reserved for generic data types;
+  * other generic parameters should have meaningfull names.
+
+[&#8593; back to TOC](#table-of-contents)
+
+## Typing
+
+* Don't use in-place structure definitions, create interfaces instead:
+
+  **Good**
+
+  ```ts
+  interface IPerson {
+      name: string;
+      age: number;
+  }
+
+  let person: IPerson;
+  ```
+
+  **Bad**
+
+  ```ts
+  let person: {name: string, age: number};
```

Comment:
I agree in general, but sometimes in-place declarations maybe suitable, e.g. in the following example:

```ts
interface Foo {
   inner: {
       bar: number;
   }
}
```
we don't need declare additional interface for `inner` property, because it is not used separately from `Foo` interface and we can always use `Foo['inner']` as its type.

That's why I suggest
1. Change `Don't use in-place structure definitions, create interfaces instead` to `Prefer interface to in-place structure definitions`.
2. Add some explanation.